lib/facility.rb

```diff
@@ -1,14 +1,72 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees,
+              :zipcode,
+              :open_hour,
+              :closing_hour
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(data)
+    @name = data[:name]
+    @address = data[:address]
+    @phone = data[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @zipcode = data[:zipcode]
+    @open_hour = data[:open_hour]
+    @closing_hour = data[:closing_hour]
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil unless services.include?("Vehicle Registration")
+    vehicle.registration_date = Date.today
```

Comment:
I totally think this could live here. One other thing to think about is could this live in the `Vehicle` class as a helper method that checks if it a vehicle has been registered and then set the date? But nice job!